MAYA-106376: fix undo/redo re-parenting crash due to accessing a stale prim.

HamedSabri-adsk opened the pull request:
This PR addresses the random undo/redo crash after performing multiple re-parenting operations.

ppt-adsk commented on lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp:
```diff
@@ -150,16 +154,20 @@ bool UsdUndoInsertChildCommand::insertChildUndo()
         // its subtree in the current UsdEditTarget
         {
-            auto stage = _ufeDstItem->prim().GetStage();
+            auto stage = srcPrimUsd.GetStage();
```
You sure about this?  The original code was getting the stage from the destination item to remove its prim, so I think you should be doing the same.

HamedSabri-adsk replied:
@ppt-adsk  Duh!!  addressed in 9948c423fa9ba43020d6c029f3d008dd53d01df6

ppt-adsk commented on lib/mayaUsd/ufe/UsdUndoInsertChildCommand.cpp:
```diff
@@ -124,16 +124,20 @@ bool UsdUndoInsertChildCommand::insertChildRedo()
     {
+        // we shouldn't rely on UsdSceneItem to access the UsdPrim since 
+        // it could be stale. Instead we should get the USDPrim from the Ufe::Path
+        const auto& desPrimUsd = ufePathToPrim(_ufeDstPath);
```
Same a line 164: this will give the wrong stage at line 134.

ppt-adsk replied:
Naming nit-pick: should use "dst" or "des" prefix for "destination", but not both.

ppt-adsk reviewed (changes requested):
Good catch!  In general, any command which stores UFE scene items (or USD prims, same issue) is suspicious.

I think you should simply remove the _ufeSrcItem and _ufeDstItem data members, because they are no longer useful, and doing this would completely remove the possibility of relying on stale prims.  Just use local variables where UFE scene items are needed.

HamedSabri-adsk commented:
> I think you should simply remove the _ufeSrcItem and _ufeDstItem data members, because they are no longer useful, and doing this would completely remove the possibility of relying on stale prims. Just use local variables where UFE scene items are needed.

Good point but we can't remove _ufeDstItem since it needs to be passed to insertedChild().

For now, I removed _ufeSrcItem since it serves no purpose to be stored.  Addressed in  c29d1e48169c7fd8d855b6659b7454c7ee2bcc67
